libraries/AP_HAL_Linux/RCOutput_ZYNQ.cpp

```diff
@@ -76,17 +82,25 @@ void RCOutput_ZYNQ::disable_ch(uint8_t ch)
 
 void RCOutput_ZYNQ::write(uint8_t ch, uint16_t period_us)
 {
-    if (corked) {
-        pending[ch] = period_us;
-        pending_mask |= (1U << ch);
-    } else {
-        sharedMem_cmd->periodhi[ch].hi = TICK_PER_US*period_us;
+    if (ch < MAX_ZYNQ_PWMS) {
```

Comment:
It would be good to add use a "return early on error" style here. So you just have to copy it to all functions:

```
if (ch >= PWM_CHAN_COUNT) {
    return;
}
```

See https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_Linux/RCOutput_Disco.cpp#L78 as example